addons/crm/tests/test_crm_lead.py

```diff
@@ -308,20 +319,15 @@ def test_crm_lead_partner_sync_email_phone_corner_cases(self):
         differences, wrong input, ...) """
         test_email = 'amy.wong@test.example.com'
         lead = self.lead_1.with_user(self.env.user)
-        contact = self.env['res.partner'].create({
-            'name': 'NoContact Partner',
-            'phone': '',
-            'email': '',
-            'mobile': '',
-        })
+        contact = self.contact_with_name_only
 
         lead_form = Form(lead)
         self.assertEqual(lead_form.email_from, test_email)
         self.assertFalse(lead_form.ribbon_message)
 
-        # email: False versus empty string
+        # email: False and empty strings do not trigger ribbon (nor sync)
         lead_form.partner_id = contact
-        self.assertIn('the customer email', lead_form.ribbon_message)
+        self.assertFalse(lead_form.ribbon_message)
```

Comment:
I would put these 2 lines before "# email: False and empty strings do not trigger ribbon (nor sync)"
As they are also related to the phone (from what I understand).

Maybe after you should assign a different value to the email and check that the ribbon is displayed to get the same logic as the phone?
Those 2 logics now seem differently tested although I think they share the same behavior.

Also complete the comment for phone then since you changed the one for email?